test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,646 @@
+			dat, err := ioutil.ReadFile(*serviceAccount)
+			Expect(err).NotTo(HaveOccurred())
+			fmt.Print(string(dat))
+			*serviceAccount = string(dat)
+		}
+
+		project, err = gcp.ExtractServiceAccountProjectID([]byte(*serviceAccount))
+		Expect(err).NotTo(HaveOccurred())
+		computeService, err = compute.NewService(ctx, option.WithCredentialsJSON([]byte(*serviceAccount)), option.WithScopes(compute.CloudPlatformScope))
+		Expect(err).NotTo(HaveOccurred())
+
+		extensionscluster, controllercluster = createClusters(name)
+		bastion, options = createBastion(controllercluster, name, project)
+
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cloudprovider",
+				Namespace: name,
+			},
+			Data: map[string][]byte{
+				gcp.ServiceAccountJSONField: []byte(*serviceAccount),
+			},
+		}
+
+	})
+
+	AfterSuite(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
+		}()
+
+		By("running cleanup actions")
+		framework.RunCleanupActions()
+
+		By("stopping test environment")
+		Expect(testEnv.Stop()).To(Succeed())
+	})
+
+	It("should successfully create and delete", func() {
+		By("setup Infrastructure")
+		err = prepareNewNetwork(ctx, logger, project, computeService, name, routerName, subnetName)
+		Expect(err).NotTo(HaveOccurred())
+		framework.AddCleanupAction(func() {
+			err = teardownNetwork(ctx, logger, project, computeService, name, routerName, subnetName)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		By("create namespace for test execution")
+		setupEnvironmentObjects(ctx, c, namespace(name), secret, extensionscluster)
+		framework.AddCleanupAction(func() {
+			teardownShootEnvironment(ctx, c, namespace(name), secret, extensionscluster)
+		})
+
+		By("setup bastion")
+		err = c.Create(ctx, bastion)
+		Expect(err).NotTo(HaveOccurred())
+
+		framework.AddCleanupAction(func() {
+			teardownBastion(ctx, logger, c, bastion)
+
+			By("verify bastion deletion")
+			verifyDeletion(ctx, project, computeService, options)
+		})
+
+		By("wait until bastion is reconciled")
+		Expect(extensions.WaitUntilExtensionObjectReady(
+			ctx,
+			c,
+			logger,
+			bastion,
+			extensionsv1alpha1.BastionResource,
+			10*time.Second,
+			30*time.Second,
+			5*time.Minute,
+			nil,
+		)).To(Succeed())
+
+		verifyPort22IsOpen(ctx, c, bastion)
+		verifyPort42IsClosed(ctx, c, bastion)
+
+		By("verify cloud resources")
+		verifyCreation(ctx, project, computeService, options)
+	})
+})
+
+func verifyPort22IsOpen(ctx context.Context, c client.Client, bastion *extensionsv1alpha1.Bastion) {
+	By("check connection to port 22 with netcat")
+
+	bastionUpdated := &extensionsv1alpha1.Bastion{}
+	Expect(c.Get(ctx, client.ObjectKey{Namespace: bastion.Namespace, Name: bastion.Name}, bastionUpdated)).To(Succeed())
+
+	ipAddress := bastionUpdated.Status.Ingress.IP
+	createOptions := &client.CreateOptions{}
+	err := c.Create(ctx, createNetCatPortProbeJob(bastion.Namespace, ipAddress, "22"), createOptions)
```

Comment:
why did you have to go through the hurdle and create a job that checks if the port is open? why couldn't you check it from the go code within the test?

Reply:
committed refactor in c777806b